src/roman_datamodels/datamodels/_datamodels.py

```diff
@@ -440,3 +441,74 @@ class ImageSourceCatalogModel(_RomanDataModel, _ParquetMixin):
 
 class SegmentationMapModel(_RomanDataModel):
     _node_type = stnode.SegmentationMap
+
+
+class WfiWcsModel(_RomanDataModel):
+    _node_type = stnode.WfiWcs
+
+    @classmethod
+    def from_model_with_wcs(cls, model, l1_border=4):
+        """Extract the WCS information from an exposure model post-assign_wcs
+
+        Construct a `WfiWcsModel` from any model that is used post-assign_wcs step
+        in the ELP pipeline. The WCS information is extracted out of the input model.
+        The wcs-related meta information is copied verbatim from the input model.
+
+        However, the WCS object itself is placed into the attribute 'wcs_l2'. Furthermore, a
+        modified GWCS, applicable to the Level 1 version of the input model, is created
+        and stored in the attribute 'wcs_l1'.
+
+        Parameters
+        ----------
+        model : ImageModel
+            The input data model (a WfiWcsModel will also work).
+
+        l1_border : int
+            The extra border to add for the L1 wcs.
+
+        Returns
+        -------
+        wfiwcs_model : WfiWcsModel
+            The WfiWcsModel built from the input model. If the input is already
+            a WfiWcsModel, it is simply returned.
+
+        """
+        ALLOWED_MODELS = ImageModel
+
+        if isinstance(model, cls):
+            return model
+        if not isinstance(model, ALLOWED_MODELS):
+            raise ValueError(f"Input must be one of {ALLOWED_MODELS}")
+
+        # Ensure a WCS has been defined.
+        if model.meta.wcs is None:
+            raise ValueError(f"Model has no WCS defined, cannot create {cls}")
+
+        # Create base node with dummy values (for validation)
+        from roman_datamodels.maker_utils import mk_wfi_wcs
+
+        wfi_wcs = mk_wfi_wcs()
+
+        _node_update(wfi_wcs, model, only_in_to_node=True)
```

Comment:
FYI, Brett has been trying to eliminate uses of the maker utilities in the pipeline, preferring to directly create the model and then to fill out the metadata.  e.g., https://github.com/spacetelescope/romancal/pull/1675/files  The rationale is that this leave less room to forget to pass through metadata that needs to be passed through.  Does that approach work here?

Reply:
I had asked about this in the slack channel. I believe the answer is no because the use-case is a bit different. So far all the conversion methods are meant to transfer information between two, possibly very different, schemas. Hence the attributes are not necessarily one-to-one and depend on knowing what the trees are. At the moment, the only easy way of knowing what the tree looks like are the maker utils. 

Williams' and others work on the model refactoring should deal with this in a cleaner way, but that is, I believe, the only path we have.